src/sidebar/components/markdown-editor.js

```diff
@@ -134,69 +136,180 @@ ToolbarButton.propTypes = {
   label: propTypes.string,
   onClick: propTypes.func,
   shortcutKey: propTypes.string,
+  tabIndex: propTypes.string,
   title: propTypes.string,
 };
-
+/**
+ * An array of toolbar elements with a roving tab stop. Left and right
+ * array keys can be used to change focus of the elements. Home and end
+ * keys will navigate to the first and last elements respectively.
+ *
+ * Canonical example
+ * https://www.w3.org/TR/wai-aria-practices/examples/toolbar/toolbar.html
+ */
 function Toolbar({ isPreviewing, onCommand, onTogglePreview }) {
+  // Total number of elements in the toolbar minus 1.
+  const maxElementIndex = 9;
+
+  // Keep track of a roving index. The active roving tabIndex
+  // is set to 0, and all other elements are set to -1.
+  const [rovingElement, setRovingElement] = useState(0);
```

Comment:
ya, that was my reason for "element" :\

okay: )

Reply:
Doing a mapping like `buttonIds` was one my initial thoughts, but it does make keeping track of the roving index more difficult.

So for example, maybe the roving index is the key into the buttonIds? But its not obvious which one is next or prev.  The simple js object does not preserve order in every case and even if it did in this, its probably not a good standard to assume order so I would want to use the `Map` object or keep track of a secondary ordering array -- more complex.  Which is why I dumped the index values directly into the JSX -- simple and dumb, but works. Then I wrote that test that asserts the order in case it was ever tampered.

Reply:
>  The simple js object does not preserve order in every case and even if it did in this, its probably not a good standard to assume order so I would want to use the Map object or keep track of a secondary ordering array -- more complex

By "does not preserve order" are you referring to the iteration order over keys in an object? It used to be the case that this was theoretically implementation-dependent. In practice all browsers ended up making it deterministic because code relied on it and this order was [made part of the ES6 spec](https://2ality.com/2015/10/property-traversal-order-es6.html#traversing-the-own-keys-of-an-object). Incidentally the same is true of Python dicts, but that only changed much more recently.

Reply:
Awe, didn't know that about ES6 changes. Also, I Think I was overthinking this as the values in that buttonsId serves as the order, duh, and the rovingIndex can remain largely the same as its implemented.